crypto/evp/e_chacha20_poly1305.c

```diff
@@ -216,16 +215,21 @@ static int chacha20_poly1305_init_key(EVP_CIPHER_CTX *ctx,
     actx->mac_inited = 0;
     actx->tls_payload_length = NO_TLS_PAYLOAD_LENGTH;
 
-    /* pad on the left */
-    memset(temp, 0, sizeof(temp));
-    if (actx->nonce_len <= CHACHA_CTR_SIZE)
-        memcpy(temp + CHACHA_CTR_SIZE - actx->nonce_len, iv, actx->nonce_len);
```

Comment:
Instead of reindenting everything, a more minimal change would be:
```c
if (iv == NULL) {
  chacha_init_key(ctx, inkey, NULL, enc);
  return 1;
}